site/src/components/MdxPageContent.tsx

```diff
@@ -38,12 +38,30 @@ export const MdxPageContent: VFC<MdxPageContentProps> = ({
     setAnchor(router.asPath.split("#")[1] ?? "");
   }, [router]);
 
+  const scrolledOnce = useRef(false);
+
   useEffect(() => {
     const headingWithCurrentAnchor = headings.find(
       (heading) => heading.anchor === anchor,
     );
 
-    if (headingWithCurrentAnchor) {
+    const previousRoute = sessionStorage.getItem("previousRoute");
+
+    const shouldScrollToAnchor =
+      // if anchor is empty and we haven't scrolled, prevent it
+      (anchor === "" && scrolledOnce.current) ||
+      // if anchor is not empty, always allow scroll
+      anchor !== "" ||
+      // OR if previous path is either a docs or spec page
+      (previousRoute?.includes("/docs") && router.asPath?.includes("/docs")) ||
+      (previousRoute?.includes("/spec") && router.asPath?.includes("/spec"));
```

Comment:
I think you just need this condition and not the ref or anything else - what is the other stuff for?

Reply:
The first condition that I've added here is to enable the behavior seen in point 1 of this comment: https://github.com/blockprotocol/blockprotocol/pull/179#issuecomment-1028647755

Reply:
Understood - we don't need to scroll to the top if switching inside /docs or inside /spec - there is no new information at the top.

Reply:
@CiaranMn sorry but I don't think I've understood this comment well. Do you mean that this PR is good to go if I remove the 1st condition, ie, `(anchor === "" && scrolledOnce.current)` from this list?

Reply:
I mean we don't need the behavior you describe: "Clicking on the header of a different page in the sidenav scrolls you to the top." - clicking on the header of a different page in the sidenav should scroll you to the header.

We should only scroll to the very top if arriving at /docs from a non-docs page, or arriving at /spec from a non-spec page: not when moving within /docs or within /spec.

Reply:
Ahh okay, that will probably require more changes within the codebase. Will have a look.

Reply:
It's the same as the current behavior, except scrolling to the very top if moving within /docs or within /spec - I think you're almost there as you can tell if we're moving within /docs or within /spec